### DISCUSSION TITLE
Am I building MSLS correctly?

### DISCUSSION THREAD

**Discussion by edwinbradford** [lloc/Multisite-Language-Switcher#235] [Q&A] [answered]
I can't seem to find any docs explaining how to build Multisite-Language-Switcher locally. I'm not a PHP developer but I've looked through the project and got it working with a lot of help from ChatGPT.

This is what I recall doing so far:

- Install php8.1-fpm
- Install php8.1-curl
- Install php8.1-xml
- Install Composer
- Install phpDocumentor
- Install npm (using nvm)

Run:

    $ composer build

Run:

    $ bash bin/git-release.sh

The last command generates the plugin and a .zip of the plugin. It looks like npm is only required to generate JavaScript, CSS and documentation assuming that's what `package.json` is for, none of which I need to modify. I've attempted to build the `docs` directory using phpDocumentor but they appear to be largely auto generated and I can't find anything explaining how to build the project.

In `composer.json` there don't appear to be any build scripts so is that it and all I need to do is run `git-release.sh` or am I missing anything important?

Incidentally if I run `composer analyze` I get 172 errors before it fails with error code 1 at `vendor/bin/phpstan`.

**Comment by lloc** [comment, marked as answer]
Sorry for the late answer. To be honest, I never opened the discussion tab before.

The current version can be found here: https://wordpress.org/plugins/multisite-language-switcher/

If you use the plugin in an environment where you have the repository installed, then you would just call:

`composer install --no-dev`

I use PHPStan to improve steadily the code quality. Errors are expected in this case.

**Reply by edwinbradford** [reply]
I never normally use this tab either, I can't believe that's all I was missing after getting so far, oh well it was a good learning experience. You fixed the issue I was working on a few weeks ago so everything is working perfectly thanks.
